pyfstat/core.py

```diff
@@ -520,6 +520,8 @@ def _get_SFTCatalog(self):
             " with constraints {}.".format(self.sftfilepattern, ", ".join(constr_str))
         )
         self.SFTCatalog = lalpulsar.SFTdataFind(self.sftfilepattern, constraints)
+        if self.SFTCatalog.length == 0:
+            raise IOError("No SFTs found.")
```

Comment:
I just checkted [XLALSFTdataFind](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/_s_f_tcatalog_8c_source.html#l00068).
 
Is this here because XLALSFTdataFind does not check whether the final set of SFTs is non-empty? (It checks many other things, but not that one I'd say).

Reply:
Yes, during testing I ran into some case where it was throwing a rather ugly error further down without this check.